pytorch_lightning/plugins/precision/precision_plugin.py

```diff
@@ -0,0 +1,108 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import math
+from typing import Generator, Union
+
+import torch
+from torch.optim import Optimizer
+
+from pytorch_lightning.core import LightningModule
+from pytorch_lightning.plugins.base_plugin import Plugin
+
+
+class PrecisionPlugin(Plugin):
+    EPSILON = 1e-6
+    precision = 32
+
+    def master_params(self, optimizer: torch.optim.Optimizer) -> Generator[torch.Tensor, None, None]:
+        """The master params of the model. Returns the plain model params here.
+        Maybe different in other precision plugins.
+
+        """
+        for group in optimizer.param_groups:
+            for p in group["params"]:
+                yield p
+
+    def connect(self, model: torch.nn.Module, optimizers, lr_schedulers):
+        """Connects this plugin to the accelerator and the training process"""
+        return model, optimizers, lr_schedulers
+
+    def backward(
+        self,
+        model: LightningModule,
+        closure_loss: torch.Tensor,
+        optimizer: torch.optim.Optimizer,
+        opt_idx: int,
+        should_accumulate: bool,
```

Comment:
`should_accumulate` isn't used anywhere here?

Reply:
For now this was leaved as I remembered there was an open PR using that. We definitely have to iterate after we merged all of these and do a final cleanup